### PR TITLE
"Auto-fill on page load" options

### PR DESCRIPTION
**Pull request by eliykat**
These are the changes to jslib required for this feature.

Refer to the main pull request here: https://github.com/bitwarden/browser/pull/1455

### CODE REVIEW

**Comment by eliykat** [comment]
This is ready for final review for merging.

**Review by cscharf — DISMISSED** [review]

**Comment by eliykat** [comment]
Just had to cause and then resolve some conflicts

File: src/models/data/loginData.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export class LoginData {
     password: string;
     passwordRevisionDate: string;
     totp: string;
+    autofillOnPageLoad: boolean;
```
Comment (cscharf) [inline src/models/data/loginData.ts:11]: Does it matter that your server model is `nullable` here?

File: src/models/domain/login.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ export class Login extends Domain {
     password: EncString;
     passwordRevisionDate?: Date;
     totp: EncString;
+    autofillOnPageLoad: boolean;
```
Comment (cscharf) [inline src/models/domain/login.ts:17]: Same, nullabe?

File: src/models/data/loginData.ts (hunk 1 above)
Comment (eliykat) [reply src/models/data/loginData.ts:11]: It's intentional - `true` is autofill, `false` is don't autofill, `null` is treated as "not configured (therefore use default settings)". I originally had an enum for this, but I thought that was overkill when a `bool` would do the job.

File: src/models/domain/login.ts (hunk 2 above)
Comment (eliykat) [reply src/models/domain/login.ts:17]: See explanation above

File: src/models/data/loginData.ts (hunk 1 above)
Comment (cscharf) [reply src/models/data/loginData.ts:11]: Yeah, for whatever reason I was thinking you had to define `boolean | null` but you don't, so yeah, it's my dumb 🤦🏻

**Review by cscharf — APPROVED** [review]

**Pull request merged**